### ISSUE TITLE
1.1.9 - Search with incomplete words using wildcard options (now Advanced Search 1.2.16)

### ISSUE THREAD

**Issue by janahoffmann** [gbhl/bhl-europe#41]
Search using wildcards, e.g. [], *, ?

Related bug #179

**Comment by janahoffmann** [comment]
ready for testing

wildcard option \* and ? work fine
[] not accepted

**Comment by JiriFrank** [comment]
Bug: using [] reporting an error. Brackets are needed for search for ranges mentioned in Advance search3: [start TO end]

**Comment by JiriFrank** [comment]
Testing of feature 1.1.9 - Search with incomplete words using wildcard options (*)

Feature description from the Catalogue of user requirements:

[], *, ?

Testers:
@AntonioGVH
@fwelter
@heimor
@janahoffmann
@JFTester
@JiriFrank
@LarissaS

**Comment by JiriFrank** [comment]
Testing of feature 1.1.9 - Search with incomplete words using wildcard options (*)

Feature description from the Catalogue of user requirements:

[], *, ?

Testers:
@fwelter

**Comment by LarissaS** [comment]
Searching with \* (mollus_, spin_) works Ok, but results are presented without logic? Why not first show best match in the title or author and then the rest (with matches inside the bibliographic description)?

Searching for "Anders_" gives 32 results, after sorting it on CREATOR I've got next (see screen print). Not understandable why two records there "Anders_" is in the abstract are sitting in between records where Anderson is an author?

Searching for "Boule_" and "Boul_" gives O result, while with "boul_" - 50, "boule_"- 42 and "boulenger" - 39, searching with "Boulenger" gives also 39 results

Search with [] gives error

Search with ? gives no results

-----Original Message-----

**Comment by AntonioGVH** [comment]
It seems that it works properly, if the beggining of the word is included.

Queriing by 'evo_' gets a lot of results related with evolution
Queriing by 'volu_' does seems to get results with 'evolution', but gets results with 'volume'

Antonio

Quoting JiriFrank:

**Comment by AntonioGVH** [comment]
I forgot to say that it would be interesting to include a summary of results. E.g.

-queriing by 'evo*' the results could be: evolution -23; evolutionary  32; etc

Quoting JiriFrank:

**Comment by fwelter** [comment]
1.1.9 wildcards \* ? []

"*antides du muse"
- no result

"?antides du muse"
- no result

"mantid\* du muse"
- returns 1 result (mantides du Musée)

"mantid? du muse"
- returns 1 result (mantides du Musée)

"manti\* du muse"
- no result

"manti? du muse"
- no result

antide
- no result

*antide
- no result

?antide
- no result

manti*
- the number of hits was not indicated, you have to count "Data
  Provider" (or "Rights"): 24 + 4 + 1. 
  1. Oberösterreichisches Landesmuseum (24)
  2. NCB Naturalis (4)
  3. UB der Humboldt-Universität, Zweigbibliothek Museum für
     Naturkunde (1)
- timeout error 4 times

manti?
- the number of hits was not indicated, you have to count "Data
  Provider" (or "Rights"): 3 + 1. 
  1. Oberösterreichisches Landesmuseum (2)
  2. NCB Naturalis (1)
- timeout error 3 times

mantid*
- the number of hits was not indicated, you have to count "Data
  Provider" (or "Rights"): 11 + 1. 
  1. Oberösterreichisches Landesmuseum (11)
  2. NCB Naturalis (1)

mantid?
- the number of hits was not indicated, you have to count "Data
  Provider" (or "Rights"): 11 + 1. Oberösterreichisches Landesmuseum (5)

mantida*
- 5 results

mantida?
- no results

mantidae
- 5 results

*antidae
- error 400

?antidae
- error 400

*antida
- error 400

_antid_
- error 400

_antid_
- error 400

*usee
- error 400

?usee
- error 400

basalt*
- the number of hits was not indicated, you have to count "Data
  Provider" (or "Rights"): 16 + 9 + 6. 

"basalt*"
- the number of hits was not indicated, you have to count "Data
  Provider" (or "Rights"): 9 + 6 + 6. 

basalt?
- 3 results (among these was 1 result that had Basalt? in the title)

"basalt?"
- the number of hits was not indicated, you have to count "Data
  Provider" (or "Rights"): 9 + 6 + 6. 

[mantid], [antid], [mantida], mantid], [mantidae]
- all gave error 400
- works only at the end, not at the beginning of the word

? works only at the end, not at the beginning of the word, but works
somehow differently from *

[ ] did not work (but I never use this and have no experience with it)

---
- when there were many results the page took long time to load in and
  blocked the mouse of my computer. this was because all the thumbnails
  needed to load in. it seems that also those thumbnails needed to load
  in that were not displayed on the screen. if I had installed to
  display 25 results in the hitlist and there were 1500 results the
  thumnails for all 1500 results loaded in. after that the computer was
  largely blocked and very slow, I tried to close the window as soon as
  possible and I also tried not to select search terms that produced so
  many results.
- nowhere was displayed how many titles were found in total, usually I
  listed the numbers given for data providers. only maximum 5 data
  providers were shown, I discovered this bug only very late in the
  testing process.
- it is also possible that Date, Language and the other facet lists
  had also restrictions for 5 lines to be shown only. I realised this
  only after 2 weeks testing.
- the big problem was that too many results were shown, mainly for two
  reasons:

1 - the result lists generally displayed titles where none of the
words were contained in the title, author or year, but in the text of
the abstract and notes of the item page. 

2 - single journal volumes were displayed for serial runs, they
blocked all intelligent search attempts and spammed the results list
with over and over the same journal title. 
- it was generally very difficult (better say, practically impossible)
  to test for completeness of results. restricting search for (1) only
  in title, author and date and (2) exclude journal metadata of single
  journal volume item pages (and instead create one single journal title
  hit) would probably solve the problems.

Francisco

---

F. Welter-Schultes
Zoologisches Institut, Berliner Str. 28, D-37073 Goettingen
Phone +49 551 395536, Fax +49 551 395579
http://www.gwdg.de/~fwelter
http://www.animalbase.org

**Comment by akohlbecker** [comment]
searching with wildcards \* and ? is now possible in advanced search using the [] is no longer needed since we have an UI for time spans and intervals
